module/account/src/Domain/ValueObject/Email.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See license.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\Account\Domain\ValueObject;
+
+use Egulias\EmailValidator\EmailValidator;
+use Egulias\EmailValidator\Validation\RFCValidation;
+
+/**
+ */
+class Email
+{
+    /**
+     * @var string
+     */
+    private $value;
+
+    /**
+     * @param string $value
+     */
+    public function __construct(string $value)
+    {
+        $value = mb_strtolower(trim($value));
+
+        if (!self::isValid($value)) {
+            throw new \InvalidArgumentException('Value is not correct email');
+        }
+
+        $this->value = $value;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValue(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @return string
+     */
+    public function __toString(): string
+    {
+        return $this->value;
+    }
+
+    /**
+     * @param Email $value
+     *
+     * @return bool
+     */
+    public function isEqual(Email $value): bool
+    {
+        return $value->getValue() === $this->value;
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return bool
+     */
+    public static function isValid(string $value): bool
+    {
+        return (new EmailValidator())->isValid($value, new RFCValidation());
```

Comment:
If you use `mb_strtolower(trim($value))` in constructor you need to us it in this method to